controllers/managedocs_controller.go

```
@@ -255,3 +323,104 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespaceName, configmap)
+	if err == nil {
+		labels := configmap.Labels
+		if _, ok := labels[r.DeleteConfigMapLabelKey]; !ok {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() (reconcile.Result, error) {
+
+	r.Log.Info("Deleting subscription")
+	if result, err := r.deleteResource(r.Client, &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{Name: r.AddonSubscriptionName, Namespace: r.namespace},
+	}, r.AddonSubscriptionName); err != nil {
+		return ctrl.Result{}, err
+	} else if !result.IsZero() {
+		return result, nil
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(context.Background(), csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return ctrl.Result{}, err
+	}
+	for _, csv := range csvList.Items {
+		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
+			deployerCsvName := csv.Name
+			if result, err := r.deleteResource(r.Client, &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{Name: deployerCsvName, Namespace: r.namespace},
+			}, deployerCsvName); err != nil {
+				return ctrl.Result{}, err
+			} else if !result.IsZero() {
+				return result, nil
+			}
+		}
+	}
+	r.Log.Info("CSV removed successfully")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ManagedOCSReconciler) isComponentReady() bool {
+	subComponent := r.managedOCS.Status.Components
+
+	if subComponent.StorageCluster.State != v1.ComponentReady {
+		return false
+	}
+	return true
+}
+
+func (r *ManagedOCSReconciler) isPvcRemoved() (bool, error) {
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err := r.UnrestrictedClient.List(context.Background(), pvcList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list PVCs")
+		return false, err
```

Please return a proper err and not the general one, for example:
```
return false, fmt.ErrorF(Unable to list PVCs: %v, err)
```

If you do that I am not even sure we will need the log.Error any more